lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -354,6 +354,52 @@ static bool hoistNonSideEffectExpr(Operation *op) {
   return true;
 }
 
+/// This function create a new wire for the operand of probe op so that
+/// we can attach the name for the probed value.
+static void replaceSingleOperandOfProbeOpWithWire(Block &block,
+                                                  ProbeOp probeOp) {
+  if (probeOp.getNumOperands() != 1)
+    return;
+  bool isProceduralRegion = block.getParentOp()->hasTrait<ProceduralRegion>();
+
+  auto operand = probeOp.getOperand(0);
+  // If we are probing WireOp, then skip.
+  if (isa_and_nonnull<WireOp>(operand.getDefiningOp()))
+    return;
+
+  // Find a block to create a placeholder wire.
+  Block *wireBlock = &block;
+
+  // TODO: Consider whether we should care about the procedural region case
+  //       because it is tedious to create a placeholder wire.
+  if (isProceduralRegion)
+    wireBlock = findParentInNonProceduralRegion(probeOp)->getBlock();
+
+  ImplicitLocOpBuilder builder(probeOp.getLoc(), probeOp.getContext());
+  builder.setInsertionPointAfterValue(operand);
+
+  // Read inout values implicitly.
+  if (operand.getType().isa<sv::InOutType>())
+    operand = builder.create<ReadInOutOp>(operand);
+
+  // Create a new wire to associate with ProbeOp.
+  // TODO: Add an useful name to wire.
+  // TODO: Don't create a new wire if the operand can define symbols.
+  builder.setInsertionPointToStart(wireBlock);
+  auto newWire = builder.create<WireOp>(operand.getType());
```

Comment:
I probably had to attach some symbol to this wire so that the wire has a global name